janus_core/calculations/geom_opt.py

```diff
@@ -114,7 +116,8 @@ def __init__(
         symmetrize: bool = False,
         symmetry_tolerance: float = 0.001,
         angle_tolerance: float = -1.0,
-        filter_func: Callable | str | None = FrechetCellFilter,
+        filter_class: Callable | str | None = FrechetCellFilter,
+        filter_func: Callable | str | None = None,
```

Comment:
Do you want to relegate `filter_func` to a `**kwargs` so it's not visible in the function hinting?

Reply:
```python
if extras := kwargs.keys() - {"filter_func", "model_path"}:
    raise NameError("Invalid kwargs ({', '.join(extras)}) passed")
```
Maybe not though.

Reply:
What I was going to do was `kwargs.pop` (defaulting to `None`), then raise against a non-empty `kwargs`, since I want to keep how they're used/passed around otherwise, so the actual variables need to be set, but I think it's probably more trouble then it's worth.